src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/VideoLinkMigrationService.kt

```diff
@@ -0,0 +1,144 @@
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkBookingEventType
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkAppointmentRepository
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkBookingEventRepository
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkBookingRepository
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+@Service
+class VideoLinkMigrationService(
+  private val videoLinkBookingRepository: VideoLinkBookingRepository,
+  private val videoLinkBookingEventRepository: VideoLinkBookingEventRepository,
+  private val videoLinkAppointmentRepository: VideoLinkAppointmentRepository,
+) {
+  companion object {
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getVideoLinkBookingToMigrate(videoBookingId: Long): VideoBookingMigrateResponse {
+    val videoLinkBooking = videoLinkBookingRepository.findById(videoBookingId)
+      .orElseThrow { EntityNotFoundException("Video link booking ID {videoBookingId} was not found") }
+
+    log.info("Migrating video link booking ID $videoBookingId")
+
+    // Get the history of events for this booking
+    val eventEntities = videoLinkBookingEventRepository.findEventsByVideoLinkBookingId(videoBookingId)
+    require(eventEntities.isNotEmpty()) { "Video link booking ID $videoBookingId has no events" }
+
+    // Is this a cancelled booking? i.e. containing a DELETE event.
+    val cancelledBooking = eventEntities.any { it.eventType == VideoLinkBookingEventType.DELETE }
+
+    // Get the appointments linked to the booking - cancelled bookings will have none (they get removed)
+    val appointments = videoLinkAppointmentRepository.findAllByVideoLinkBooking(videoLinkBooking)
+
+    if (!cancelledBooking) {
+      require(appointments.isNotEmpty()) { "Video link booking ID $videoBookingId has no appointments" }
+      require((appointments.size <= 3)) { "Video link booking ID $videoBookingId has more than 3 appointments" }
+    }
+
+    // Extract the related appointments - will be null if a cancelled booking otherwise reflect the appointments
+    val main = appointments.find { it.hearingType == HearingType.MAIN }
+    val pre = appointments.find { it.hearingType == HearingType.PRE }
+    val post = appointments.find { it.hearingType == HearingType.POST }
+
+    // Extract appointment details from the latest CREATE or UPDATE event if this booking is cancelled
+    val appointmentsFromEvent = extractAppointmentsFromEvent(eventEntities, cancelledBooking)
+
+    if (!cancelledBooking) {
+      require(main != null) { "Video link booking ID $videoBookingId has no main appointment" }
+    }
+
+    return VideoBookingMigrateResponse(
+      videoBookingId = videoLinkBooking.id!!,
+      offenderBookingId = videoLinkBooking.offenderBookingId,
+      courtCode = videoLinkBooking.courtId?.let { videoLinkBooking.courtId } ?: "UNKNOWN",
```

Comment:
Don't need the `courtCode = videoLinkBooking.courtId?.let { videoLinkBooking.courtId } ?: "UNKNOWN"`

Can just do `courtCode = videoLinkBooking.courtId ?: "UNKNOWN"`

Reply:
Done